src/sample.jl

```diff
@@ -436,13 +447,8 @@ function mcmcsample(
 
             Distributed.@async begin
                 try
-                    Distributed.@sync for (i, _rng, _model, _sampler) in
-                                          zip(1:nchunks, rngs, models, samplers)
-                        chainidxs = if i == nchunks
-                            ((i - 1) * chunksize + 1):nchains
-                        else
-                            ((i - 1) * chunksize + 1):(i * chunksize)
-                        end
+                    Distributed.@sync for (chainidxs, _rng, _model, _sampler) in
+                                          zip(chain_index_groups, rngs, models, samplers)
```

Comment:
AFAICT you could compute the chunks on the fly:
```suggestion
                    Distributed.@sync for (i, _rng, _model, _sampler) in
                                          zip(interval, rngs, models, samplers)
                        chainidx_lo = (i - 1) * m + min(n, i - 1) + 1
                        chainidx_hi = chainidx_lo + m - (i > n)
                        chainidxs = chainidx_lo:chainidx_hi
```

Reply:
Hmm, yes, we could. I think subtracting i > n is a bit too smart for me, though. Let me write up something similar...

Reply:
Ok, let me know what you think 🙂